### ISSUE TITLE
Use entry_points instead of scripts in setup.py

### ISSUE THREAD

**Issue by blais** [beancount/beancount#155]
**[Original report](https://bitbucket.org/blais/beancount/issue/155) by rkhwaja (Bitbucket: [rkhwaja](https://bitbucket.org/rkhwaja), GitHub: [rkhwaja](https://github.com/rkhwaja)).**

----------------------------------------

On windows, the install puts files named like "bean-bake" in the python scripts directory. To run these on windows you have to type:
py.exe <<PYTHON_INSTALL_DIR>>\scripts\bean-bake

If you use entry_points, it will generate the right files for windows and others.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Tried it.

First, works with setuptools, but it creates other (ugly) files under /bin... like this:


```
#!python

/tmp/beancount-setup-test.vblc9o7i/bin/bean-check
/tmp/beancount-setup-test.vblc9o7i/bin/pyrsa-encrypt-bigfile
/tmp/beancount-setup-test.vblc9o7i/bin/pyrsa-decrypt
/tmp/beancount-setup-test.vblc9o7i/bin/bean-web
/tmp/beancount-setup-test.vblc9o7i/bin/chardetect
/tmp/beancount-setup-test.vblc9o7i/bin/bean-report
/tmp/beancount-setup-test.vblc9o7i/bin/pyrsa-priv2pub
/tmp/beancount-setup-test.vblc9o7i/bin/pyrsa-keygen
/tmp/beancount-setup-test.vblc9o7i/bin/pyrsa-encrypt
/tmp/beancount-setup-test.vblc9o7i/bin/pyrsa-decrypt-bigfile
/tmp/beancount-setup-test.vblc9o7i/bin/pyrsa-verify
/tmp/beancount-setup-test.vblc9o7i/bin/pyrsa-sign
/tmp/beancount-setup-test.vblc9o7i/bin/bean-query
...

```
That's on Ubuntu.


Second, does not work without setuptools.

I could condition the option to use scripts= when setuptools isn't there, but I'm concerned this is going to create more complexity and problems. I'm not sure.

Given the problems above, I could also make setup() use entry_points only under Windows? This might handle it.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Actually, here's what I can do without breaking other things: I can use entry_points only when (1) Windows is present, and (2) setuptools is installed. That should leave everything else untouched. If you know packaging well - and it sounds like you might - I'd appreciate some feedback.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #155: Use entry_points instead of scripts in setup.py.
